### PR TITLE
Polish set_optimizer Interface

### PR DESCRIPTION
**Pull request by Aurelius84**
+ Modify `set_optimizer(optim, loss_name)`  into `set_optimizer(optim, index_of_loss)`
```python
adam = fluid.optimizer.AdamOptimizer(learning_rate=0.001)
# set optimizer	         
program_translator = ProgramTranslator()
program_translator.set_optimizer(adam, index_of_loss=1)
```

### CODE REVIEW

File: python/paddle/fluid/dygraph/dygraph_to_static/program_translator.py [hunk 1]
```diff
@@ -261,19 +263,20 @@ def __init__(self, exe=None, place=None):
...
         self._optimizer = None
-        self._already_minimized = False
+        self._loss_name = None
         # Once main_program is changed, should run startup_program.
         self._need_startup = True
 
     def get_output(self, dygraph_func, *args, **kwargs):
         """
-        Returns the output tensors for dygraph function and its arguments
+        Return the output tensors for dygraph function and its arguments
```
Comment (zhhsplendid) [inline python/paddle/fluid/dygraph/dygraph_to_static/program_translator.py:274]: Returns, supports is more common in SWE, but you don't have to change in this PR

**Review by zhhsplendid — APPROVED** [review]
LGTM

**Review by liym27 — APPROVED** [review]
LGTM

File: python/paddle/fluid/dygraph/dygraph_to_static/program_translator.py (hunk 1 above)
Comment (Aurelius84) [reply python/paddle/fluid/dygraph/dygraph_to_static/program_translator.py:274]: I will change it back in https://github.com/PaddlePaddle/Paddle/pull/23652

**Pull request merged**